internal/onboarding/aws_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package onboarding_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	subject "go.mondoo.com/cnspec/v11/internal/onboarding"
+)
+
+func TestGenerateAwsHCL_KeyBased(t *testing.T) {
+	code, err := subject.GenerateAwsHCL(subject.AwsIntegration{
+		Name:      "test-key-integration",
+		Space:     "space-123",
+		AccessKey: "AKIAXXXXXXXXXXXXXXXX",
+		SecretKey: "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+	})
+	assert.Nil(t, err)
+
+	expected := `terraform {
+  required_providers {
+    mondoo = {
+      source  = "mondoohq/mondoo"
+      version = "~> 0.19"
+    }
+  }
+}
+
+provider "mondoo" {
+  space = "space-123"
+}
+
+resource "mondoo_integration_aws" "this" {
+  credentials = {
+    key = {
+      access_key = "AKIAXXXXXXXXXXXXXXXX"
+      secret_key = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
```

Comment:
I don't think we want to write these secrets to a file 🤔 I am wondering if we should, instead, create Terraform variables and then pass these values via environment variables like `TF_VAR_secret_key` or so.

Reply:
Nope. Those are ok to be on the generated file.